### DISCUSSION TITLE
sheet:自定义命令里如何触发刷新

### DISCUSSION THREAD

**Discussion by raintoway** [dream-num/univer#1844] [Q&A | 常见问题]
sheet:自定义命令里如何触发刷新

**Comment by jikkai** [comment]
请仔细阅读 [Facade API 文档](https://univer.ai/zh-cn/guides/facade/)

**Reply by raintoway** [reply]
[Facade API 文档](https://univer.ai/zh-cn/guides/facade/) 这个文档是看过的，很遗憾，里面没有关于主动触发sheet渲染的方式。以及自定义命令里也没有关于主动触发sheet渲染的方式

**Reply by raintoway** [reply]
我也看过具体的setRangeValuesMutation的实现，里面有对底层model的更改，但是这个更改是利用消息订阅/观察者还是其他方式来触发渲染函数的，这块没找到对应的代码，并且这种方式的trigger/fire是很难调试的，就像blocksuite/AFFiNE那样
